In [ ]:
import pandas as pd

# [Dataset 1](https://www.kaggle.com/datasets/praveengovi/emotions-dataset-for-nlp)

In [ ]:
df1 = pd.read_csv("../data/raw/extra_dataset_Praveen.csv")
df1.head()

,text,label
0,i didnt feel humiliated,sadness
1,i can go from feeling so hopeless to so damned...,sadness
2,im grabbing a minute to post i feel greedy wrong,anger
3,i am ever feeling nostalgic about the fireplac...,love
4,i am feeling grouchy,anger


In [ ]:
df1 = df1[df1['label'].isin(['love', 'surprise'])]
df1.head()

,text,label
3,i am ever feeling nostalgic about the fireplac...,love
6,ive been taking or milligrams or times recomme...,surprise
9,i feel romantic too,love
32,i have seen heard and read over the past coupl...,surprise
47,i can t let go of that sad feeling that i want...,love


In [ ]:
df1['label'].value_counts()

,count
label,
love,1641
surprise,719


We're not getting a lot more instances but anything is better than nothing

# [Dataset 2](https://www.kaggle.com/datasets/pashupatigupta/emotion-detection-from-text)

In [ ]:
df2 = pd.read_csv("../data/raw/extra_dataset_PashupatiGupta.csv")
df2.head()

,tweet_id,sentiment,content
0,1956967341,empty,@tiffanylue i know i was listenin to bad habi...
1,1956967666,sadness,Layin n bed with a headache ughhhh...waitin o...
2,1956967696,sadness,Funeral ceremony...gloomy friday...
3,1956967789,enthusiasm,wants to hang out with friends SOON!
4,1956968416,neutral,@dannycastillo We want to trade with someone w...


In [ ]:
df2 = df2.drop('tweet_id', axis=1)
df2 = df2.rename(columns={'sentiment': 'label', 'content': 'text'})
df2.head()

,label,text
0,empty,@tiffanylue i know i was listenin to bad habi...
1,sadness,Layin n bed with a headache ughhhh...waitin o...
2,sadness,Funeral ceremony...gloomy friday...
3,enthusiasm,wants to hang out with friends SOON!
4,neutral,@dannycastillo We want to trade with someone w...


In [ ]:
df2 = df2[df2['label'].isin(['love', 'surprise', 'enthusiasm'])]
df2.head()

,label,text
3,enthusiasm,wants to hang out with friends SOON!
14,surprise,Got the news
16,love,@annarosekerr agreed
56,enthusiasm,"bed...sorta. today was good, sara has strep th..."
94,love,@RobertF3 correct! I ADORE him. I just plucked...


In [ ]:
df2['label'].value_counts()

,count
label,
love,3842
surprise,2187
enthusiasm,759


In [ ]:
df2['label'] = df2['label'].replace('enthusiasm', 'surprise')
df2['label'].value_counts()


,count
label,
love,3842
surprise,2946


## Merging df1 and df2...

In [ ]:
df_merged = pd.concat([df1, df2], ignore_index=True)
df_merged.head()

,text,label
0,i am ever feeling nostalgic about the fireplac...,love
1,ive been taking or milligrams or times recomme...,surprise
2,i feel romantic too,love
3,i have seen heard and read over the past coupl...,surprise
4,i can t let go of that sad feeling that i want...,love


In [ ]:
df_merged['label'].value_counts()

,count
label,
love,5483
surprise,3665


# [Dataset 3](https://www.kaggle.com/datasets/simaanjali/emotion-analysis-based-on-text/data)

In [ ]:
df3 = pd.read_csv("../data/raw/extra_dataset_SimaAnjali.csv")
df3.head()

,Unnamed: 0,text,Emotion
0,0,i seriously hate one subject to death but now ...,hate
1,1,im so full of life i feel appalled,neutral
2,2,i sit here to write i start to dig out my feel...,neutral
3,3,ive been really angry with r and i feel like a...,anger
4,4,i feel suspicious if there is no one outside l...,neutral


In [ ]:
df3.columns

Index(['Unnamed: 0', 'text', 'Emotion'], dtype='object')

In [ ]:
df3 = df3.drop('Unnamed: 0', axis=1)
df3 = df3.rename(columns={'Emotion': 'label'})

In [ ]:
df3.head()

,text,label
0,i seriously hate one subject to death but now ...,hate
1,im so full of life i feel appalled,neutral
2,i sit here to write i start to dig out my feel...,neutral
3,ive been really angry with r and i feel like a...,anger
4,i feel suspicious if there is no one outside l...,neutral


In [ ]:
df3 = df3[df3['label'].isin(['love', 'surprise'])]
df3.head()

,text,label
5,i feel jealous becasue i wanted that kind of l...,love
74,i feel like she has taken on the role of a gra...,love
171,i feel like im back to the arms of a beloved l...,love
356,i am feeling so festive right now and not just...,love
380,i don t discuss even my feelings for beloved w...,love


In [ ]:
df3['label'].value_counts()

,count
label,
love,39553
surprise,6954


In [ ]:
df_merged = pd.concat([df_merged, df3], ignore_index=True)
df_merged.head()

,text,label
0,i am ever feeling nostalgic about the fireplac...,love
1,ive been taking or milligrams or times recomme...,surprise
2,i feel romantic too,love
3,i have seen heard and read over the past coupl...,surprise
4,i can t let go of that sad feeling that i want...,love


In [ ]:
df_merged['label'].value_counts()


,count
label,
love,45036
surprise,10619


In [ ]:
df_merged.drop_duplicates(subset=['text'], inplace=True)
df_merged['label'].value_counts()

,count
label,
love,22776
surprise,6974


# [Dataset 4](https://www.kaggle.com/datasets/bryanhuerta/social-media-sentiment-data)

In [ ]:
df4 = pd.read_csv("../data/raw/extra_dataset_BryanHuerta.csv")
df4.head()

,ID_Column,Text,Sentiment,Time_Column,User,Platform,Hashtags,Retweets,Likes,Country,Year,Month,Day,Hour
0,17,Late-night gaming session with friends.,Positive,2023-01-20 00:05:00.000000 UTC,GamerX,Twitter,#Gaming #LateNight,18,35,UK,2023,1,20,0
1,327,"Fearful eyes scanning the shadows, a prisoner...",Fearful,2023-01-07 03:30:00.000000 UTC,NightPrisoner,Twitter,#Fearful #LurkingMonsters,18,36,India,2023,1,7,3
2,336,"Loneliness, a silent companion in the night, ...",Loneliness,2019-09-19 03:30:00.000000 UTC,NocturnalEcho,Facebook,#Loneliness #SolitaryNocturne,18,36,Australia,2019,9,19,3
3,305,"Eyes wide open in the night, fearful shadows ...",Fearful,2022-05-12 02:00:00.000000 UTC,NightPrisoner,Facebook,#Fearful #ImaginedHorrors,18,36,India,2022,5,12,2
4,510,"At the summit of Mount Fuji, a breathtaking su...",Accomplishment,2020-04-15 05:30:00.000000 UTC,FujiSummitAchiever,Facebook,#Accomplishment #SunriseHues,28,55,Japan,2020,4,15,5


In [ ]:
df4 = df4[['Text', 'Sentiment']]
df4 = df4.rename(columns={'Text': 'text', 'Sentiment': 'label'})

In [ ]:
unique_labels = df4['label'].unique()
print(', '.join(unique_labels))

Positive, Fearful, Loneliness, Accomplishment, Hopeful, Awe, Anticipation, Motivation, Despair, Negative, Gratitude, Empowerment, Love, Overwhelmed, Curiosity, Proud, InnerJourney, Mindfulness, Disgust, Confusion, Anxiety, Frustrated, Calmness, Neutral, Disappointed, Blessed, Anger, Arousal, Contentment, Grateful, Optimism, LostLove, Elation, Ambivalence, Apprehensive, Bitterness, Determination, Jealous, Excitement, Regret, Compassionate, Free-spirited, Adoration, Acceptance, Radiance, Betrayal, Happiness, Envy, Jealousy, Euphoria, Kind, Admiration, Ruins, Heartbreak, Satisfaction, Devastated, Inspiration, Resentment, Amusement, Shame, Surprise, Bad, Hate, Joy, Thrill, Adrenaline, Happy, Kindness, Desolation, Enthusiasm, Exploration, Relief, Confident, Appreciation, Freedom, Amazement, DreamChaser, Exhaustion, Helplessness, Numbness, Dismissive, Frustration, Pride, Nostalgia, Empathetic, Rejuvenation, Compassion, Inspired, Sad, Whimsy, Grief, Adventure, Desperation, Serenity, Sadness, 

Awe, Gratitude, Love, Confusion, Arousal, Adoration, Admiration, Thrill, Adrenaline, Wonder, Energy, Suspense


In [ ]:
labels_to_keep = ['Awe', 'Gratitude', 'Love', 'Confusion', 'Arousal', 'Adoration', 'Admiration', 'Thrill', 'Adrenaline', 'Wonder', 'Energy', 'Suspense']
df4_copy = df4[df4['label'].isin(labels_to_keep)].copy()
df4_copy['label'].value_counts()

,count
label,
Gratitude,17
Awe,9
Confusion,8
Arousal,4
Thrill,4
Admiration,4
Love,3
Wonder,3
Adoration,2


In [ ]:
labels_to_display = ['Wonder', 'Adrenaline', 'Arousal', 'Thrill', 'Suspense']
filtered_df = df4_copy[df4_copy['label'].isin(labels_to_display)]
for index, row in filtered_df.iterrows():
    print(f"{row['text']}, {row['label']}")

Arousal of excitement before a much-awaited trip., Arousal
Arousal of excitement as I await a special announcement., Arousal
Arousal of excitement for an upcoming adventure., Arousal
Arousal of excitement for an upcoming festival., Arousal
Embracing the thrill of speed on a rollercoaster's exhilarating twists., Thrill
Riding the adrenaline rush on a rollercoaster's wild twists., Adrenaline
Thrilled to witness the grandeur of a cultural festival., Thrill
In the carnival of emotions, the rollercoaster of thrill sends the heart on a wild ride. , Thrill
In the midst of the Amazon rainforest, a symphony of wildlife creates an orchestra of nature's wonders. , Wonder
Experiencing the thrill of a high-speed Formula 1 race, fans are on the edge of their seats as drivers compete for the championship. , Thrill
Binge-watching a thrilling crime series, the suspense keeps the viewer on the edge of their seat, creating a rush of adrenaline. , Suspense
Awe-inspired by the vastness of the cosmos on a s

In [ ]:
filtered_df = df4_copy[df4_copy['label'].isin(['Awe'])]
for index, row in filtered_df.iterrows():
    print(f"{row['text']}, {row['label']}")

Awe-struck by the breathtaking sunrise over the mountains., Awe
Awe-struck by the grandeur of an ancient cathedral's intricate architecture., Awe
Awe-inspired by the grandeur of an ancient cathedral's intricate architecture., Awe
In the embrace of the Grand Canyon, nature's masterpiece, a moment to marvel at Earth's grandeur. , Awe
At the golf tournament, the golfer's precision and focus lead to a stunning victory, leaving spectators in awe of their exceptional skill. , Awe
Walking the Great Wall of China, each step a testament to ancient engineering marvels. , Awe
Experiencing awe at the breathtaking sunset., Awe
Awe-struck by the beauty of the night sky., Awe
At the summit of Machu Picchu, a breathtaking panorama that whispers the secrets of ancient civilizations. , Awe


In [ ]:
labels_to_surprise = ['Surprise', 'Awe', 'Confusion', 'Arousal', 'Thrill', 'Adrenaline', 'Wonder', 'Energy', 'Suspense']
labels_to_love = ['Gratitude', 'Love', 'Adoration', 'Admiration']

In [ ]:
df4_copy['label'] = df4_copy['label'].replace(labels_to_surprise, 'surprise')
df4_copy['label'] = df4_copy['label'].replace(labels_to_love, 'love')
df4_copy['label'].value_counts()

,count
label,
surprise,31
love,26


In [ ]:
df_merged = pd.concat([df_merged, df4_copy], ignore_index=True)
df_merged['label'].value_counts()

,count
label,
love,22802
surprise,7005


In [ ]:
df_merged = pd.read_csv("../data/raw/dataset_6labels.csv")
df_merged.head()

# Custom dataset for surprise

In [ ]:
# Load more_surprise.csv
df_surprise = pd.read_csv("../data/raw/more_surprise.csv")
df_surprise.head()

,count
label,
surprise,663


Yay we have 663 more instances for `surprise`

Let's merge everything now and save it:

In [ ]:
df_merged = pd.concat([df_merged, df_surprise], ignore_index=True)
df_merged['label'].value_counts()

,count
label,
love,22802
surprise,7668


In [ ]:
df_merged.to_csv('data/raw/love_surprise_bonus.csv', index=False)
print("CSV file created successfully!")

CSV file created successfully!
